controllers/operatorpolicy_controller.go

```diff
@@ -929,55 +1132,172 @@ func (r *OperatorPolicyReconciler) handleInstallPlan(
 	return updateStatus(policy, installPlanApprovedCond(approvedVersion), relatedInstallPlans...), nil
 }
 
+func (r *OperatorPolicyReconciler) mustnothaveInstallPlan(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	ownedInstallPlans []unstructured.Unstructured,
+) ([]metav1.Condition, bool, error) {
+	relatedInstallPlans := make([]policyv1.RelatedObject, len(ownedInstallPlans))
+
+	if policy.Spec.RemovalBehavior.ApplyDefaults().InstallPlan.IsKeep() {
+		for i := range ownedInstallPlans {
+			relatedInstallPlans[i] = leftoverObj(&ownedInstallPlans[i])
+		}
+
+		return nil, updateStatus(policy, keptCond("InstallPlan"), relatedInstallPlans...), nil
+	}
+
+	for i := range ownedInstallPlans {
+		relatedInstallPlans[i] = foundNotWantedObj(&ownedInstallPlans[i])
+	}
+
+	changed := updateStatus(policy, foundNotWantedCond("InstallPlan"), relatedInstallPlans...)
+
+	if policy.Spec.RemediationAction.IsInform() {
+		return nil, changed, nil
+	}
+
+	earlyConds := []metav1.Condition{}
+
+	if changed {
+		earlyConds = append(earlyConds, calculateComplianceCondition(policy))
+	}
+
+	for i := range ownedInstallPlans {
+		err := r.Delete(ctx, &ownedInstallPlans[i])
```

Comment:
Optional: Is there likely to be a lot of these install plans around? If so, it may be worth doing this concurrently since deletes can take a few seconds.

Reply:
OLM removes some old InstallPlans eventually, I think I read somewhere that it tries to keep a max of 5?

I don't think that waiting for several deletes in series would be a major inconvenience. I think I prefer the simplicity here.

Reply:
But this makes me wonder about situations where the `Delete` might get stuck (possibly indefinitely) waiting for finalizers or something. I think controller-runtime has a default deadline on the context it passes the Reconcile call, but I would bet it's longer than we'd really want. It might be worth looking at all of these Deletes, specifying a lower timeout, and specially reporting the status when they "get stuck". I would prefer to handle that in a separate PR.

Reply:
I don't think the delete API request gets stuck since I don't think it's blocking like `kubectl`. It does seem like the delete calls will likely be repeated every reconcile. Perhaps you can guard deletes on if the deletion timestamp is set to avoid such API requests?

Reply:
@JustinKuli what are your thoughts on this thread? Are you planning on addressing any of it in this PR or a separate PR?

Reply:
Guarding deletes if a deletion timestamp is set would be good. I think I want to do it in a separate PR.